src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -161,6 +161,153 @@ static Value builtinRegExpSearch(ExecutionState& state, Value thisValue, size_t
+        return index + 1;
+
+    return index + 2;
+}
+
+static Value builtinRegExpReplace(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(obj, RegExp, replace);
+    RegExpObject* rx = obj->asRegExpObject();
+    String* str = argv[0].toString(state);
+    Value replaceValue = argv[1];
+    size_t lengthStr = str->length();
+    bool functionalReplace = replaceValue.isCallable();
+    bool fullUnicode = false;
+    String* replacement = String::emptyString;
+    // 7. If functionalReplace is false, then
+    if (!functionalReplace) {
+        replaceValue = replaceValue.toString(state);
+    }
+
+    bool global = rx->option() & RegExpObject::Option::Global;
+
+    // 10. If global is true, then
+    if (global) {
+        fullUnicode = rx->option() & RegExpObject::Option::Unicode;
+        rx->setLastIndex(state, Value(0));
+    }
+    ValueVector results;
+
+    // 13. Repeat, while done is false
+    while (true) {
+        Value result = builtinRegExpExec(state, obj, 1, &argv[0], false);
+        // 13.c If result is null, set done to true
+        if (result.isNull()) {
+            break;
+        }
+        // 13.d
+        results.pushBack(result);
+        // 13.d.ii. If global is false, set done to true.
+        if (!global) {
+            break;
+        }
+        // 13.d.iii.
+        Value matchStr = result.asObject()->get(state, ObjectPropertyName(state, Value(0))).value(state, Value(0));
+        // 13.d.iii.3 If matchStr is the empty String, then
+        if (matchStr.toString(state)->length() == 0) {
+            size_t thisIndex = rx->computedLastIndex(state);
+            size_t nextIndex = advanceStringIndex(str, thisIndex, fullUnicode);
+            rx->setLastIndex(state, Value(nextIndex));
+        }
+    }
+    //14.
+    String* accumulatedResult = String::emptyString;
+    size_t nextSourcePosition = 0;
+
+    //16. Repeat, for each result in results,
+    for (size_t i = 0; i < results.size(); i++) {
+        Value result = results[i];
+        size_t nCaptures = result.asObject()->get(state, ObjectPropertyName(state.context()->staticStrings().length)).value(state, Value(0)).toLength(state) - 1;
+
+        // Let nCaptures be max(nCaptures − 1, 0).
+        if (nCaptures == SIZE_MAX) {
+            nCaptures = 0;
+        }
+        Value matched = result.asObject()->get(state, ObjectPropertyName(state, Value(0))).value(state, Value(0));
+        size_t matchLength = matched.toString(state)->length();
+        size_t position = result.asObject()->get(state, ObjectPropertyName(state.context()->staticStrings().index)).value(state, Value(0)).asNumber();
+
+        // Let position be max(min(position, lengthS), 0).
+        if (position > lengthStr) {
+            position = lengthStr;
+        }
+
+        size_t n = 1;
+        std::vector<String*, GCUtil::gc_malloc_ignore_off_page_allocator<String>> captures;
```

Comment:
Since this vector allocates String addresses (`String*`), so the template allocator should be assigned as `String*` too.